### ISSUE TITLE
Establishing Issue Organization Guidelines

### ISSUE THREAD

**Issue by johnrom** [jaredpalmer/formik#1533]
The goal of this organization is to 
  a) highlight discussions of importance to the community
  b) prioritize work so that maintainers can spend their limited time wisely
  c) establish exactly what work is required for a given release

This type of organization is loosely based off of open source projects like:
  - https://github.com/aspnet/AspNetCore
  - https://github.com/OrchardCMS/OrchardCore

1. Organize current issues by milestone, so that the noise of endless issues can be cancelled out. Recommended Milestones:
  v1.x hotfix: bugs or small features that should be addressed now as they affect the community
  v2.0: _only_ those issues which are prioritized for v2 launch
  backlog: bugs or features that do not need to be addressed by v2, but should be fixed after and before thinking about the ideal features in vNext. these should eventually be prioritized into v2.1, v2.2, etc milestones
  vNext: features which are important to the project, but depend on v2 launch and should not coincide with it or even be thought about regularly

2. Remove or lock `v2.0 Umbrella Issue` because it is an endless stream of weird feature requests and issues. All comments regarding v2 should belong to a specific issue or PR, which belong to the v2 milestone.

3. Standardize Labels
- I recommend removing certain types of Labels
    - Remove version tags, as they belong to milestones.
    - Remove `Component` labels (new `Focus: Components`), as they are noisy and probably will cause changes to other Components anyway. 
    - Remove Browser: Chrome and similar labels, as these are more implementation details that should be written under the issue.
- Most other labels should be prefixed
- Discussion: When an issue is attempting to be prioritized by the community.
- Priority: Establish the importance of an issue to a milestone
  - Required (High, Medium)
  - Icing on the Cake / Optional: Those things which could be kicked to the next milestone, if warranted.
- Focus: Establish the general area of work (similar to current Components)
  - Components (currently Fastfield, Field, FieldArray, Form, Formik)
  - TypeScript
  - React Native
  - Documentation
  - Examples
  - Testing
- Status: 
  - Needs Browser Testing
  - Completed (currently "done")
  - In Progress
  - Help Wanted
- Resolution (existing)
- Type (existing)
  - maybe remove Umbrella because that should be established by Milestones
  - maybe keep Umbrella but keep it locked and as a pointer to the milestone, since milestones aren't always obvious to the community on Github
  - add `Type: Cross-Browser inconsistency` if you want to preserve "Browser: Chrome" like labels
- Other (no prefix)
  - Stale
  - Good First issue
  - Pinned

4. Add this information to the Contributing docs and establish guidelines for successfully prioritizing issues using labels and milestones.

**Comment by Andreyco** [comment]
Awesome idea!

**Comment by jaredpalmer** [comment]
Sounds good.

**Comment by Andreyco** [comment]
Are there any actions to be done done? The sooner this is done, the better. At least some of those points

**Comment by johnrom** [comment]
I think I'll have some time this weekend to get this kickstarted, though I would like to know if you guys have any revisions / alternate ideas on anything before I get started

**Comment by Andreyco** [comment]
I am seeing a lot of v2 related Issues open - whether they are issues of ideas, doesnt matter. To get v2 shipped, we should focus on those first and this would be great starting point - having a timeline, bucket of deal breaking issues and ideas which we can deliver later (new features does not need to ship with 2.0)

As part of this, we should go thru issues and triage them. And clean/merge issue labels.

**Comment by johnrom** [comment]
Agreed. The goal is to get _every_ issue into a milestone, so that it's clear when an issue hasn't been looked into yet.

I think the first step is finalizing the content of this issue, and adding it to the `.github/CONTRIBUTING.md` file so that current and future maintainers and community members understand the process, and revisions to this process can be version controlled.

**Comment by jaredpalmer** [comment]
Don’t make a mess 😂😂

**Comment by jaredpalmer** [comment]
But yeah this is fine

**Comment by johnrom** [comment]
If I have time to implement this weekend, I will not perform any changes that don't exist already. I will rename labels except those that remain the same, and make sure those updated labels apply to the issues as they currently exist. I will move anything marked `[v2]` or labelled `2.0` into the v2.0 milestone, but I'll leave everything else unprioritized as I'm not knowledgeable enough about the codebase to decide how important most of these items are to the community.

**Comment by jaredpalmer** [comment]
Sounds awesome! You rock!

**Comment by johnrom** [comment]
I had a thought about this today as some issues were pouring in. Some issues are more like "questions regarding v2" and not necessarily actions that need to be taken toward the milestone. So I think we should remove the "2.0" label (since it denotes a specific version), but add a "v2" label referring to the general API around v2 as it's materializing. The label will allow us to go back and see an overview of questions related to the v2 API, and repeated questions will give us information about what API design can be improved.

I'm also wondering if it might be better to switch the Stale bot to only react to issues with a specific label (or set of labels), like "awaiting author response", since long running issues are often just waiting on external dependencies (like v2). It seems like sometimes automatically closed issues cause more work because somebody has to comment and then we have to re-open them. Some automatically closed issues might even be issues that the maintainers expect to work on, but now won't see on deck because it was automatically closed.

**Comment by johnrom** [comment]
I had some time today, so I did the following:

- Issues from label 2.0 which I determined to belong to v2.0 instead of the `v2` api label were moved to the v2.0 milestone.
- I renamed the label `2.0` to `v2` to mean questions and issues surrounding the new API, and not necessarily the version `2.0`.
- Renamed "Browser: Chrome" to `Type: Browser Inconsistency`
- Renamed "TypeScript" to "Focus: TypeScript"
- Locked v2 Umbrella Issue #828 in support of the Milestone

I held off on renaming things like `help wanted` or `done` in case they have special meaning to the github repository. I'll need @jaredpalmer 's feedback on whether that's the case.

I'm going to pause before doing more to get your feedback on how the new system is working. Let me know!

**Comment by sw-yx** [comment]
not stale
